iocore/net/SSLSNIConfig.cc

```diff
@@ -193,13 +194,15 @@ SNIConfig::startup()
 void
 SNIConfig::cloneProtoSet()
 {
+  SCOPED_MUTEX_LOCK(lock, naVecMutex, this_ethread());
```

Comment:
Why does this need to be locked? It's not modifying the vector, only class local data.

Reply:
I just searched for all occurrences of naVec and added lock to all of it.

Reply:
I still think it should be locked since there might be a read and write at the same time.